lib/src/touch_spin.dart

```diff
@@ -68,40 +68,38 @@ class _TouchSpinState extends State<TouchSpin> {
           padding: widget.iconPadding,
           iconSize: widget.iconSize,
           color: minusBtnDisabled
-              ? widget.iconDisabledColor ?? Theme.of(context).disabledColor
-              : widget.iconActiveColor ??
-                  Theme.of(context).textTheme.button.color,
+              ? widget.iconDisabledColor
+              : widget.iconActiveColor,
           icon: widget.subtractIcon,
           onPressed: minusBtnDisabled
               ? null
               : () {
                   num newVal = _value - widget.step;
-                  setState(() {
-                    _value = newVal;
-                  });
-                  if (widget.onChanged != null) widget.onChanged(newVal);
+
+                  setState(() => _value = newVal);
+
+                  widget.onChanged(newVal);
                 },
         ),
         Text(
-          '${widget.displayFormat == null ? _value.toString() : widget.displayFormat.format(_value)}',
+          '${widget.displayFormat == null ? _value.toString() : widget.displayFormat?.format(_value)}',
```

Comment:
Using the `!` operator is appropriate since you know `displayFormat` is not-null.

However, you could also simply say:
```dart
widget.displayFormat?.format(_value) ?? _value.toString()
```